src/transaction_builder.js

```diff
@@ -447,7 +461,8 @@ TransactionBuilder.prototype.sign = function (vin, keyPair, redeemScript, hashTy
     input.prevOutScript !== undefined &&
     input.pubKeys !== undefined &&
     input.signatures !== undefined &&
-    input.signatures.length === input.pubKeys.length
+    input.signatures.length === input.pubKeys.length &&
+    input.pubKeys.length > 0
```

Comment:
if `input.pubKeys` is `0`, we should try to expand whatever we can for more information